docs/spec/spec.md

```diff
@@ -57,7 +59,45 @@ Each client represents a pool of connections to the database. The pool of connec
 public isolated function init(string account_identifier, string user, string password, Options? options = (), sql:ConnectionPool? connectionPool = ()) returns sql:Error?;
 ```
 
-**Configurations available for initializing the Snowflake client:**
+Additionally, the Snowflake connector has an Advanced client, which allows the user to use other authentication methods such as public/private key authentication, etc. Right now, the Advanced client supports only Basic authentication and Key based authentication, but it can be extended to support other authentication methods in the future.
+The Advanced client can be initialized as follows:
```

Comment:
The following lines represent the definition of the advanced client rather than the client initialization